src/imars3d/backend/workflow/engine.py

```diff
@@ -13,17 +13,24 @@
 from typing import Any, Optional
 
 
+class WorkflowEngineExitCodes(Enum):
+    r"""Exit codes to be used with workflow engine errors."""
+
+    SUCCESS = 0
+    ERROR_GENERAL = 1
+    ERROR_VALIDATION = 2
+
+
 class WorkflowEngineError(RuntimeError):
     """Base class for workflow engine errors."""
 
-    pass
+    exit_code = WorkflowEngineExitCodes.ERROR_GENERAL.value
```

Comment:
:1st_place_medal: 
I didn't know one could attach exit code to custom exceptions. This is quite useful

Reply:
This is the real reason behind all the nonsense. It is a member variable! Define whatever you like.